src/main.rs

```diff
@@ -45,6 +45,18 @@ impl Default for DiagnosticArgs {
     }
 }
 
+/// Set of Parameters used to specify the extra options for the `weaver` command.
+/// The WeaverArgs will be shared across all commands. So only the general options should be
+/// included here.
+#[derive(Args, Debug)]
+pub struct WeaverArgs {
```

Comment:
I suggest renaming this struct to `CommonRegistryArgs` and moving it to `src/registry/mod.rs`, as it will encapsulate the common arguments for all the `registry` subcommands. For global arguments (e.g., --debug), Clap already provides a mechanism. However, that mechanism cannot be used here because `--follow-symlinks` does not apply to commands unrelated to the registry.